src/annotations/_AnnotationManager.js

```diff
@@ -0,0 +1,224 @@
+import dcmjs from "dcmjs";
+
+import _MarkupManager from "./markups/_MarkupManager";
+import MeasurementMarkup from "./markups/measurement";
+import ArrowMarker from "./markers/arrow";
+import FreeTextMarkup from "./markups/freeText";
+import Enums from "../enums";
+import { getUnitsSuffix } from "./markups/utils";
+
+const { Marker, Markup, FeatureEvents } = Enums;
+
+class _AnnotationManager {
+  constructor({ map, source, controls, getROI } = {}) {
+    this.props = { map, source, controls, getROI };
+
+    /** Markups */
+    this[Markup.Measurement] = MeasurementMarkup(this.props);
+    this[Markup.FreeTextEvaluation] = FreeTextMarkup(this.props);
+
+    /** Markers */
+    this[Marker.Arrow] = ArrowMarker(this.props);
+
+    this.props.markupManager = new _MarkupManager({
+      map,
+      source,
+      formatters: {
+        [Marker.Arrow]: this[Marker.Arrow].format,
+        [Markup.Measurement]: this[Markup.Measurement].format,
+        [Markup.FreeTextEvaluation]: this[Markup.FreeTextEvaluation].format,
+      },
+      onDrawStart: this.onDrawStart.bind(this),
+      onDrawEnd: this.onDrawEnd.bind(this),
+    });
+  }
+
+  /**
+   * Gets the code meaning of a given measurement or evaluation content item.
+   *
+   * @param {NumContentItem} contentItem The measurement or evaluation content item
+   */
+  getContentItemCodeMeaning(contentItem) {
+    const { ConceptNameCodeSequence } = contentItem;
+    return ConceptNameCodeSequence.length
+      ? ConceptNameCodeSequence[0].CodeMeaning
+      : ConceptNameCodeSequence.CodeMeaning;
+  }
+
+  /**
+   * Add or update a ROI measurement.
+   *
+   * @param {Feature} feature The feature
+   * @param {NumContentItem} measurement The measurement content item
+   * @param {string} codeMeaning The code meaning
+   */
+  addOrUpdateMeasurement(feature, measurement, codeMeaning) {
```

Comment:
Logic extracted from manager and reimplemented, thanks

Reply:
I would suggest passing `code` rather than `codeMeaning` (see [this comment](https://github.com/MGHComputationalPathology/dicom-microscopy-viewer/pull/43/files/bb0b14ac99a3ed3bc08dc5da0a5f24ad99ee76aa#r584430833))